src/connectors/menu/__tests__/connectMenu-test.js

```diff
@@ -404,4 +404,117 @@ describe('connectMenu', () => {
       expect(firstRenderingOptions.canToggleShowMore).toBe(false);
     });
   });
+
+  describe('routing', () => {
+    const getInitializedWidget = () => {
+      const rendering2 = jest.fn();
+      const makeWidget2 = connectMenu(rendering2);
+      const widget = makeWidget2({
+        attributeName: 'category',
+      });
+
+      const helper = algoliasearchHelper(
+        { addAlgoliaAgent: () => {} },
+        '',
+        widget.getConfiguration({})
+      );
+      helper.search = sinon.stub();
+
+      widget.init({
+        helper,
+        state: helper.state,
+        createURL: () => '#',
+        onHistoryChange: () => {},
+      });
+
+      const { refine } = rendering2.mock.calls[0][0];
+
+      return [widget, helper, refine];
+    };
+
+    describe('getWidgetState', () => {
+      test('should give back the object unmodified if there are no refinements', () => {
```

Comment:
After seeing many tests that are written in the same way (unmodified, add an entry, do not override namespace entries), I wonder if it would have been feasible to write **one** getWidgetState utility with configuration parameters and unit test that one instead, delegating calls to every widgets afterwards and only checking how we call it from widgets.

Reply:
Some tests are identical indeed. But not all, do you have a proposal for making both special tests and generic ones?

Reply:
After more carefully looking at `getWidgetState` implementations indeed it seems pretty different between widgets and not something easily made generic so all good